pkg/devfile/parser/data/v2/commands_test.go

```diff
@@ -121,26 +173,23 @@ func TestDevfile200_GetCommands(t *testing.T) {
 			}
 
 			commands, err := d.GetCommands(tt.filterOptions)
-			if !tt.wantErr && err != nil {
-				t.Errorf("TestDevfile200_GetCommands() unexpected error - %v", err)
-				return
-			} else if tt.wantErr && err == nil {
-				t.Errorf("TestDevfile200_GetCommands() expected an error but got nil %v", commands)
-				return
-			} else if tt.wantErr && err != nil {
+			if (err != nil) != tt.wantErr {
+				t.Errorf("TestDevfile200_GetCommands() error = %v, wantErr %v", err, tt.wantErr)
 				return
-			}
+			} else if err == nil {
+				assert.Equal(t, len(tt.wantCommands), len(commands), "expected length not the same as returned length")
 
-			for _, wantCommand := range tt.wantCommands {
-				matched := false
 				for _, devfileCommand := range commands {
-					if wantCommand == devfileCommand.Id {
-						matched = true
+					matched := false
+					for _, wantCommand := range tt.wantCommands {
+						if wantCommand == devfileCommand.Id {
```

Comment:
`wantCommands`  is always a subset of `commands`. `commands` is all the devfile commands & `wantCommands` is whatever we want from that list - `commands`/devfile commands, so  `wantCommands` can be everything or a subset depending on DevfileOptions.

it works fine when `wantCommands` == `commands` which means there is no filter or filter returned everything and this is test case number 1 by default for all the Get functions

```
{
			name: "Get the necessary commands",
			currentCommands: []v1.Command{
				{
					Id: "command1",
					CommandUnion: v1.CommandUnion{
						Exec: &v1.ExecCommand{},
					},
				},
				{
					Id: "command2",
					CommandUnion: v1.CommandUnion{
						Composite: &v1.CompositeCommand{},
					},
				},
			},
			filterOptions: common.DevfileOptions{},
			wantCommands:  []string{"command1", "command2"},
			wantErr:       false,
		},
```

Reply:
I think the other way of doing it would be:

```
var gotCommands []string
for _, command := range commands {
	gotCommands = append(gotCommands, command.Id)
}

assert.Equal(t, tt.wantCommands, gotCommands, "expected length not the same as returned length")
```

Reply:
I did't get it. `commands` is not all the devfile commands, it is filtered commands as it was defined by
```
commands, err := d.GetCommands(tt.filterOptions)
```

If the filter does not function as expected, there is case that the filtered commands contains more element than `wantCommands`

And I don't think you can do `assert.Equal` to compare `wantCommands` and gotCommands`, since slice in golang is a struct with order.  I did not try, but I would expect different element order will be considered as `NotEqual` in golang.

Reply:
updated, I check length before looping thru and return when there is a length mismatch.